backend/index.js

```diff
@@ -121,7 +128,12 @@ app.get("/post", async (req, res, next) => {
       .populate("author", ["username"])
       .sort({ createdAt: -1 })
       .limit(20);
-    res.json(posts);
+    res.json(
+        posts.map((post) => ({
+            ...post.toJSON(),
+            cover: `/uploads/${post.cover}`
```

Comment:
A good practice here is to not let the frontend know where we're storing the files on our backend. IT IS the responsibilty of the backend to provide the full path of the file, either the full URL or just the pathname.